frontend/src/components/common/form/AutoCompleteText.tsx

```diff
@@ -5,32 +5,33 @@ import { Form } from 'react-bootstrap';
 import { useState } from 'react';
 import { DisplayError } from './DisplayError';
 import { useFormikContext } from 'formik';
+import { SelectOption } from './Select';
 
-export const AutoCompleteText: React.FC<any> = ({ field, options, placeholder }) => {
-  let items: any[] = [];
+export type IAutoCompleteProps = {
+  field: string;
+  placeholder?: string;
+  options: SelectOption[];
+};
 
+export const AutoCompleteText: React.FC<IAutoCompleteProps> = ({ field, options, placeholder }) => {
   const { setFieldValue, handleChange } = useFormikContext<any>();
 
-  options.forEach((option: { label: any; value: any }) => {
-    items.push(option);
-  });
-
-  const [suggestions, setSuggestions] = useState<any>([]);
-  const [text, setText] = useState<any>('');
+  const [suggestions, setSuggestions] = useState<SelectOption[]>([]);
+  const [text, setText] = useState<string>('');
 
-  const onTextChanged = (e: any) => {
+  const onTextChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
     const val = e.target.value;
-    let dynamicSuggestions: any[] = [];
+    let dynamicSuggestions: SelectOption[] = [];
```

Comment:
Is `SelectOption` an object or an interface?  If it's an interface it should be renamed.